### PR TITLE
Reorg some pkgs

### PR DESCRIPTION
**Pull request by hugelgupf**
See commit messages.

### CODE REVIEW

File: pkg/strace/internal/abi/abi_linux.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ import (
 	"syscall"
```
Comment (rminnich) [inline pkg/strace/internal/abi/abi_linux.go]: why internal? the abi package struck me as something that could be generally useful.

File: pkg/mount/switch_root_linux.go [hunk 2]
```diff
@@ -2,13 +2,10 @@
 // Use of this source code is governed by a BSD-style
```
Comment (rminnich) [inline pkg/mount/switch_root_linux.go]: We thought we needed this for nichrome, but never used it. Is there a case to be made for deleting it?

File: pkg/loop/loop_linux.go [hunk 3]
```diff
@@ -2,22 +2,79 @@
...
+	Mounted bool
+
+	// dir is the directory the block device was mounted on.
+	dir string
+}
+
+// New initializes a Loop struct and allocates a loodevice to it.
+//
+// source is the file to use as a loop block device. target is the directory
+// the device should be mounted on.
```
Comment (rminnich) [inline pkg/loop/loop_linux.go]: target?

File: pkg/loop/loop_linux.go [hunk 4]
```diff
@@ -2,22 +2,79 @@
...
+}
+
+// Mount mounts the provided source file, with type fstype, and flags and data options
+// (which are usually 0 and ""), using the allocated loop device.
+func (l *Loop) Mount(path string, flags uintptr) error {
+	l.dir = path
+	if err := unix.Mount(l.Dev, path, l.FSType, flags, l.Data); err != nil {
+		return err
+	}
+	l.Mounted = true
```
Comment (rminnich) [inline pkg/loop/loop_linux.go]: Isn't this a bit dicey? Some rando can umount it behind your back and then this is not going to be correct. I'm not sure why it is needed.

File: pkg/loop/loop_linux.go [hunk 5]
```diff
@@ -2,22 +2,79 @@
...
+}
+
 const forceUnmount = unix.MNT_FORCE | unix.MNT_DETACH
 
 // Unmount unmounts and frees a loop. If it is mounted, it will try to unmount it.
 // If the unmount fails, we try to free it anyway, after trying a more
-// forceful unmount. We don't log errors, but we do return a concatentation
-// of whatever errors occur.
+// forceful unmount.
 func (l *Loop) Unmount(flags int) error {
```
Comment (rminnich) [inline pkg/loop/loop_linux.go]: and here I see where you use Mounted but it's a small guarantee. I wonder why we created Mounted in the first place. It's problematic to say the least.

**Comment by hugelgupf** [comment]
I already opened a NiChrome change for wpa.

Re abi - golang.org/x/sys/unix *is* that. We should be using that instead of abi.

File: pkg/loop/loop_linux.go (hunk 4 above)
Comment (hugelgupf) [reply pkg/loop/loop_linux.go]: This is something I didn't change from the existing loop code, just moved it - yeah, it kinda is

**Comment by hugelgupf** [comment]
Fair, I'll open some different PRs

File: pkg/mount/switch_root_linux.go (hunk 2 above)
Comment (hugelgupf) [reply pkg/mount/switch_root_linux.go]: No, FB uses switch_root.

Comment (rminnich) [reply pkg/mount/switch_root_linux.go]: wow, that's good to hear.

File: pkg/loop/loop_linux.go (hunk 4 above)
Comment (rminnich) [reply pkg/loop/loop_linux.go]: file an issue and let it go in as is?

**Comment by hugelgupf** [comment]
Split out #1313, #1314, #1315.

**Pull request merged**